### PR TITLE
[catmem] Move pending_ops & pack_result to runtime

### PR DESCRIPTION
**Pull request by anandbonde**
This pr moves the pending_ops & pack_result to runtime.

### CODE REVIEW

File: src/rust/catmem/queue.rs [hunk 1]
```diff
@@ -109,7 +102,6 @@ impl SharedCatmemQueue {
             }
         }
         self.ring.prepare_closed()?;
-        self.cancel_pending_ops(Fail::new(libc::ECANCELED, "this queue was closed"));
```
Comment (ppenna) [inline src/rust/catmem/queue.rs:112]: Where is cancelletaion happening now? I missed this :)

**Review by ppenna — APPROVED** [review]
LGTM

File: src/rust/catmem/queue.rs (hunk 1 above)
Comment (anandbonde) [reply src/rust/catmem/queue.rs:112]: This now happens at a centralized place in this function: https://github.com/microsoft/demikernel/blob/2fa835968cee1db9025a5e41efe2bc80236f19a1/src/rust/runtime/mod.rs#L228.

This change went in earlier in another PR. :)

Comment (ppenna) [reply src/rust/catmem/queue.rs:112]: Yes, that is what I was thinking. Thank you for the reference :)

**Pull request merged**